tests/unit/clients/python/test_on_err.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+import pytest
+from google.protobuf.pyext._message import RepeatedCompositeContainer
+
+from jina.excepts import BadClientCallback
+from jina.flow import Flow
+from jina.proto import jina_pb2
+
+
+@pytest.mark.parametrize('cb_on, x_type', [('DOCS', RepeatedCompositeContainer),
+                                           ('GROUNDTRUTHS', RepeatedCompositeContainer),
+                                           ('REQUEST', jina_pb2.Request),
+                                           ('BODY', jina_pb2.Request.IndexRequest),
+                                           ])
+def test_diff_field(cb_on, x_type):
+    def validate(x):
+        isinstance(x, x_type)
```

Comment:
you need to add an assert? or the assert will make the index fail and should fail?